### ISSUE TITLE
Number of output features in classification layers of fine tuned model

### ISSUE THREAD

**Issue by YashMishra96** [kexinhuang12345/clinicalBERT#6]
I was trying to use the model/discharge_readmission for predicting readmission on the same MIMIC III dataset, however when I was trying to use:
 model = BertForSequenceClassification.from_pretrained('model/discharge_readmission', num_labels=2), there was an error :
_RuntimeError: Error(s) in loading state_dict for BertForSequenceClassification:
	size mismatch for classifier.weight: copying a param with shape torch.Size([1, 768]) from checkpoint, the shape in current model is torch.Size([2, 768]).
	size mismatch for classifier.bias: copying a param with shape torch.Size([1]) from checkpoint, the shape in current model is torch.Size([2])._

So I guess the number of features in the classification layer is just 1, can someone please confirm this. Also how should I interpret this single output when trying to calculate the accuracy?

**Comment by kexinhuang12345** [comment]
Hi, 

Yes, you should use num_labels = 1 for binary outcome. As we are interested in a binary outcome (readmission or not), we output one value and feed it into a sigmoid function which will generate a value from 0-1 measuring the likelihood of readmission. Then, you can use binary entropy loss, and use ROC-AUC or PR-AUC from sklearn to evaluate. I hope this clarifies!
